### DISCUSSION TITLE
BPP implementation of Ratings API: request for clarification

### DISCUSSION THREAD

**Discussion by sudhirrg** [beckn/mobility#39] [Q&A]
The Rating api, just gives a provision to specify a rating,but does not indicate a specific rating scale (e.g. 0 to 5 with 5 being the highest rating).How do we specify a range and the meaning of each value?
   The on_rating api,does not have a field to send back an appropriate feedback message as response.However it says provide a url of a feedback form using xInput.
   If there is no feedback form in the application and just accepts the rating and send an appropriate on_rating message saying "Thank you for your rating", how do we send the rating and the message?

**Comment by ravi-prakash-v** [comment]
The rating range is usually specified as part of the network policy. The meaning of each value is also set at a network level. 

The idea is to introduce a common rating standard across a network. Typically it is 1-5 but it may not always be the case. 


However, in case the BPP wants to send their own rating range with custom meaning, it is not supported in the spec. 

In case the BPP does want to collect additional information, it can send an optional `XInput` object in the `on_rating` request body. This is nothing but a standard HTML/XML form hosted at the BPP's end that can be rendered at the BAP. For more information regarding the `XInput` schema, click [here](https://github.com/beckn/protocol-specifications/blob/master/docs/BECKN-007-The-XInput-Schema.md)

In case the BPP does not have a feedback form, it can simply "not call" the `on_rating` callback endpoint.

However, I feel it is a good idea to introduce a rating acknowledement message in the `on_rating` endpoint. Please feel free to send a PR that adds a "rating_ack_message" to the `on_rating` API request body. That will make the `rating / on_rating` flow a little symmetric.
